src/debugpy/server/adapters.py

```diff
@@ -346,6 +350,40 @@ def next_request(self, request: Request):
         self._tracer.step_over(thread)
         return {}
 
+    def gotoTargets_request(self, request: Request) -> dict:
+        path = request("source", dict)["path"]
+        line = request("line", int)
+        target_id = self._goto_targets_map.obtain_key((path, line))
+        target = {"id": target_id, "label": "%s:%s" % (path, line), "line": line}
+
+        return {"targets": [target]}
+    
+
+    def goto_request(self, request: Request) -> dict:
+        thread_id = request("threadId", int)
+        thread = Thread.get(thread_id)
+        if thread is None:
+            return request.isnt_valid(f'Unknown thread with "threadId":{thread_id}')
+        target_id = request("targetId", int)
+        if target_id is None:
```

Comment:
No need for this check, since `request(..., int)` will throw the appropriate `InvalidMessageError` if the property is missing or has invalid type.

(If it really should be optional, you can do `request(..., int, optional=True)` to get an empty tuple in that case.)